### ISSUE TITLE
Automate SeqRepo updates and releases

### ISSUE THREAD

**Issue by korikuzma** [biocommons/hackathon-2023#5]
### Submitter Name

N/A

### Submitter Affiliation

N/A

### Requested By

Everyone using SeqRepo

### biocommons Repo

seqrepo

### Project Details

Automate SeqRepo updates and releases

### Skill Level

Advanced

### Required Skills

Python, Docker

**Comment by korikuzma** [comment]
@reece @andreasprlic Would you be able to provide any additional information on this?

**Comment by andreasprlic** [comment]
the seqrepo update procedure is closely tied to the UTA update procedure. Can we merge this issue with #6 ?

**Comment by korikuzma** [comment]
@andreasprlic That’s fine with me. I know these are two issues that are highly wanted. @reece are you ok with this?

**Comment by reece** [comment]
@andreasprlic in what sense do you see these as "tied"?

They are pretty different in terms of mechanism, data sources, complexity, and reliability. With the exception of his we might choose to automate the process, I don't think that lessons from one will inform the other.

So, I'd prefer to keep them separate. It's easier to compose from pieces than to disentangle a monolith.

**Comment by andreasprlic** [comment]
Currently we have a manual update procedure that includes steps for both UTA and seqrepo. When I saw this ticket to "automate" future update, I was under the impression the plan might be to wrap the steps of the manual procedure as a "workflow". As such an update of both UTA and seqrepo would be done together at the same time.

Perhaps, as we build a workflow for updating the content, we could design this so each of the steps for UTA and seqrepo could get run independently and as a separate (parallel?) process. Perhaps that would make it meet what you are expecting?

**Comment by reece** [comment]
Yes. Similar tooling for parallel workflows would be grand. Also, UTA really only depends on SeqRepo because it UTA needed to realign to get cigar strings way back before NCBI GFFs existed. With the GFFs, we don't need to realign.

**Comment by korikuzma** [comment]
@reece @larrybabb and I could co-lead if you are leading a separate project

**Comment by korikuzma** [comment]
@reece has a branch for stream updates

**Comment by korikuzma** [comment]
@reece Would you be able to fill out the `Which day(s) would the project be offered?` to help determine how many day(s) this project will take?
